lib/init.js

```diff
@@ -2,15 +2,30 @@
 
 module.exports = init
 
+var path = require('path')
 var log = require('npmlog')
 var npm = require('./npm.js')
+var npx = require('libnpx')
 var initJson = require('init-package-json')
 var output = require('./utils/output.js')
 var noProgressTillDone = require('./utils/no-progress-while-running').tillDone
 
 init.usage = 'npm init [--force|-f|--yes|-y]'
 
 function init (args, cb) {
+  if (args.length) {
+    var NPM_PATH = path.resolve(__dirname, '../bin/npm-cli.js')
+    var initerName = args[0]
+    var packageName = initerName.replace(/^(@[^/]+\/)?/, '$1create-')
+
+    var npxArgs = [process.argv0, 'npm-init', packageName, ...process.argv.slice(4)]
+    var parsed = npx.parseArgs(npxArgs, NPM_PATH)
+
+    return npx(parsed)
+      .then(cb)
+      .catch(cb)
+  }
+
```

Comment:
The promise returned by `npx` is a vanilla promise (not a bluebird one) so lacks the `nodeify` helper.
The promise `then(cb)` is executed before the `npm init` command finishes (possibly before it starts).

Reply:
The promise `then(cb)` is executed before the `npm init` command finishes which causes the process to exit.
There is a PR open on `npx` to return a promise to execute the `cb` after `npx` runs its command.
https://github.com/zkat/npx/pull/172